pkg/bmc/ipmi.go

```diff
@@ -0,0 +1,37 @@
+package bmc
+
+type ipmiAccessDetails struct {
+	bmcType  string
+	portNum  string
+	hostname string
+}
+
+func (a *ipmiAccessDetails) Type() string {
+	return a.bmcType
+}
+
+// NeedsMAC returns true when the host is going to need a separate
+// port created rather than having it discovered.
+func (a *ipmiAccessDetails) NeedsMAC() bool {
+	// libvirt-based hosts used for dev and testing require a MAC
+	// address, specified as part of the host, but we don't want the
+	// provisioner to have to know the rules about which drivers
+	// require what so we hide that detail inside this class and just
+	// let the provisioner know that "some" drivers require a MAC and
+	// it should ask.
+	return a.bmcType == "libvirt"
+}
+
+// DriverInfo returns a data structure to pass as the DriverInfo
+// parameter when creating a node in Ironic. The structure is
+// pre-populated with the access information, and the caller is
+// expected to add any other information that might be needed (such as
+// the kernel and ramdisk locations).
+func (a *ipmiAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+	return map[string]interface{}{
```

Comment:
Do the contents change sufficiently among implementations that we cannot make a struct for this return value, and likewise for the interface?

Reply:
Not really. These APIs are very "anything goes." The Ironic REST API expects a mapping as argument, where the values may be strings or other nested objects and the keys depend on the driver. To cope with the Ironic REST API, the gophercloud API takes a map[string]interface as argument, so I figured returning that here would make it easier for the caller to use it.

Reply:
:+1: